.github/workflows/build-test.yml

```
@@ -11,15 +11,15 @@ jobs:
     runs-on: ubuntu-20.04
     strategy:
       matrix:
-        node-version: [16.x, 18.x, 19.x]
+        node-version: [16.x, 18.x, 20.x]
     steps:
-    - uses: actions/checkout@v3
-      with:
-        fetch-depth: 500
+    - uses: actions/checkout@v4
+    - run: git fetch origin "${{ github.event.pull_request.base.sha }}" "${{ github.event.pull_request.head.sha }}" --depth 1
```

Doesn't the `checkout` action do the same (or a similar) fetch under the hood? Would `fetch-depth: 1` solve this problem?

---

Looks like `fetch-depth: 1` is the default.

Presumable this line was added so that the head of the base branch is fetched as well. The checkout step wouldn't otherwise fetch that.

---

Nit: I think we can omit the fetch for the head commit, since the checkout step gets that

```suggestion
    - run: git fetch origin "${{ github.event.pull_request.base.sha }}" --depth 1
```

---

I tried this but it didn't work because the head commit that gets fetched is some temporary merge commit that GitHub generates, and not the one referenced in the event. See [this](https://github.com/MetaMask/eth-phishing-detect/commit/ca05303066143df56656808091937fe6883105c7) commit for the error. Super weird but explicitly fetching the commit fixed it

---

Huh, interesting. Maybe it's checking out the branch after it's merged into the target? TIL